### PR TITLE
os/bluestore: add check for zeroed BlueFS' superblock.

### PR DESCRIPTION
**Pull request by rzarzynski**
Fixes: http://tracker.ceph.com/issues/25098
Signed-off-by: Radoslaw Zarzynski <rzarzyns@redhat.com>

### CODE REVIEW

File: src/os/bluestore/BlueFS.cc [hunk 1]
```diff
@@ -586,6 +586,19 @@ int BlueFS::_open_super()
   if (r < 0)
     return r;
 
+  if (bdev[BDEV_DB]->is_shared() && bl.is_zero()) {
```
Comment (rzarzynski) [inline src/os/bluestore/BlueFS.cc]: More complex alternative could be to e.g. extend `bluefs_super_t` with flag saying that particular instance is shadow of true super block and thus should be never used. The shadow super block would be written (in multi-bdev deployments) to `BDEV_SLOW` then.

**Comment by rzarzynski** [comment]
jenkins retest this please

**Comment by ifed01** [comment]
Now we have an ability to create/remove DB/WAL via ceph-bluestore-tool without OSD redeployment. Given that I'd prefer a general approach that keeps tracking which devices(symlinks) must exist. E.g. we can populate labels bluestore_bdev_label_t with this information. And maintain this info in-sync when devices (dis)appear on corresponding manipulations.

**Comment by ifed01** [comment]
Moreover it looks like proposed approach wouldn't handle WAL symlink loss...

**Pull request closed without merging**